pandas/_libs/src/inference.pyx

```diff
@@ -947,8 +947,13 @@ def maybe_convert_numeric(ndarray[object] values, set na_values,
     -------
     numeric_array : array of converted object values to numerical ones
     """
+
+    if len(values) == 0:
+        return np.array([], dtype='i8')
```

Comment:
this is not correct if the input is an arry, then we would want to return the same dtype (empty).